### ISSUE TITLE
Can't reset nz-rate

### ISSUE THREAD

**Issue by vellengs** [NG-ZORRO/ng-zorro-antd#713]
## Version

0.6.3

## Environment

https://ng.ant.design/#/components/rate

## Reproduction link

[https://ng.ant.design/#/components/rate](https://ng.ant.design/#/components/rate)

## Steps to reproduce

https://ng.ant.design/#/components/rate

## What is expected?

the rate can be reset.

## What is actually happening?

can't

## Other?

any way to reset the rate?

**Comment by vthinkxie** [comment]
by design, you can reset it by set ngModel = 0
https://ant.design/components/rate-cn/

**Comment by vellengs** [comment]
maybe it's a design issue. you can compare this
```
http://angular-ui.github.io/bootstrap/#!#rating
```
on ant design there have not a way to click reset 0 by default setting. 
on other hand,  if I change want to change the behavior, I have to add a extra button to do that.

**Comment by vellengs** [comment]
BTW,  any implement  for custom icon?
```
 character={<Icon type="heart" />}
```

**Comment by vthinkxie** [comment]
why is it so difficult to add a button to set ngModel=0 ? it is the same as bootstrap

**Comment by vellengs** [comment]
No， they are different, I do want to extra button; I just want to click star, if checked then reset

**Comment by vellengs** [comment]
the issue is simple, communicate is hard💔

**Comment by vellengs** [comment]
Sorry, I do not want to extra button

**Comment by vthinkxie** [comment]
You want to double click the star to reset to zero, but it is an bootstrap behavior not ant design.
ng-zorro will always keep the same behavior with ant design react version.

**Comment by vellengs** [comment]
So, there is a problem to reset, right? that was I said the issue about design.
Can you give a way to user change the behavior? 
```
 _clickRate(e: MouseEvent, index: number, isFull: boolean): void {

    // give a event to custom.
    e.stopPropagation();
    if (this._disabled) {
      return;
    }
    this._hoverValue = this._value = index + 1;
    this._hasHalf = !isFull && this._allowHalf;
    if (this._hasHalf) {
      this._value -= 0.5;
    }
    this.onChange(this._value);
  }

  _hoverRate(e: MouseEvent, index: number, isFull: boolean): void {
    e.stopPropagation();
    if (this._disabled) {
      return;
    }
    const isHalf: boolean = !isFull && this._allowHalf;
    // 如果星数一致，则不作操作，用于提高性能
    if (this._hoverValue === index + 1 && isHalf === this._hasHalf) {
      return;
    }

    this._hoverValue = index + 1;
    this._hasHalf = isHalf;
  }
```

**Comment by vthinkxie** [comment]
You can push ant design to change the default design here https://github.com/ant-design/ant-design/issues, or you can extends the component yourself.

**Comment by vellengs** [comment]
OK, I push an issue to them.

**Comment by vellengs** [comment]
Ant Design implement is on the road, When it will be implement on NZ?

**Comment by vthinkxie** [comment]
@vellengs 
will be supported after 0.7.0 released
ref #1065

**Comment by vthinkxie** [comment]
fixed in version 0.7.0
